### PR TITLE
refactor(main-doc): improve bff docs

### PR DESCRIPTION
**Pull request by zllkjc**
## Summary

1. remove the concepts of function mode and framework mode.
2. use `api/lambda/xxx` instead `api/xxx` in BFF docs.
3. remove `api/*.tsx` docs, add `api/lambda/*.tsx` docs
4. improve `sdk.mdx` and `function.mdx`

## Checklist

- [ ] I have added changeset via `pnpm run change`.
- [ ] I have updated the documentation.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: packages/document/main-doc/docs/zh/guides/advanced-features/bff/function.mdx [hunk 1]
```diff
@@ -141,22 +125,46 @@ export const post = async () => {
 
 - 可以在一个文件中定义多个不同 Method 的函数，但如果定义多个相同 Method 的函数，则只有第一个会生效。
```
Comment (yimingjfe) [inline packages/document/main-doc/docs/zh/guides/advanced-features/bff/function.mdx:126]: How about removing this line? With the method function of the same name, the editor will report an error, and this line is useless.

File: packages/document/main-doc/docs/zh/guides/advanced-features/bff/function.mdx [hunk 2]
```diff
@@ -1,10 +1,8 @@
----
-sidebar_position: 1
-title: 基础用法
----
 # 基础用法
 
-通过 Modern.js 开发的应用，可以在 `api/` 目录下定义接口函数，前端可以调用这些接口函数，即可发起请求，无需写前后端胶水层代码，同时保证前后端类型安全。
+在 Modern.js 应用中，开发者可以在 `api/` 目录下定义接口文件，并导出接口函数。在前端代码中，可以用文件引用的方式，直接调用这些接口函数，发起接口请求。
```
Comment (yimingjfe) [inline packages/document/main-doc/docs/zh/guides/advanced-features/bff/function.mdx:3]: 这样描述好像不是特别合适，现在应该是 `api/lambda`，或者 "可以在 `api` 目录下编写服务端代码"?

**Comment by yimingjfe** [comment]
LGTM!

File: packages/document/main-doc/docs/zh/guides/advanced-features/bff/function.mdx [hunk 3]
```diff
@@ -1,10 +1,8 @@
...
+
+这种调用方式我们称为**一体化调用**，开发者无需编写前后端胶水层代码，并天然的保证前后端类型安全。
```
Comment (yimingjfe) [inline packages/document/main-doc/docs/zh/guides/advanced-features/bff/function.mdx:5]:
```suggestion
这种调用方式我们称为**一体化调用**，开发者无需编写前后端胶水层代码，并天然地保证前后端类型安全。
```

File: packages/document/main-doc/docs/en/guides/advanced-features/bff/extend-server.mdx [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+In complex BFF logic, defining a server instance allows for easier organization of code logic and directory structure design through a single entry point. In this file, you can perform initialization logic, add global middleware, declare routes, and even extend the original framework.
+
+The routes defined by the BFF functions will be registered after the routes defined in the `app.ts` file. Therefore, you can also intercept the routes defined by the BFF functions here for preprocessing or early responses.
+
+:::note
+At this time, if `api/_app.ts` exists in the application, the defined middleware will be executed after the middleware exported by the `api/app.ts` instance. In most cases, middleware can cover most customization needs for BFF functions. Only when the application's server-side logic is more complex do you need to customize the server instance.
+:::
+
+:::caution
+When `api/app.ts` does not exist in the application, Modern.js will default to adding [koa-body](https://www.npmjs.com/package/koa-body). When `api/app.ts` does exist in the application, if developers wish to use BFF functions with bodies, they will need to **add koa-body themselves**.
```
Comment (yimingjfe) [inline packages/document/main-doc/docs/en/guides/advanced-features/bff/extend-server.mdx:153]:
```suggestion
When `api/app.ts` does not exist in the application, Modern.js will default to adding [koa-body](https://www.npmjs.com/package/koa-body). When `api/app.ts` does exist in the application, if developers wish to use BFF functions with bodies, they will need to **add `koa-body` themselves**.
```

File: packages/document/main-doc/docs/zh/guides/advanced-features/bff/extend-server.mdx [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+在复杂的 BFF 逻辑中，定义 Server 实例可以更方便通过一个入口来组织代码逻辑，设计目录结构。在这个文件中，可以执行初始化逻辑，添加全局中间件，声明路由，甚至扩展原有框架。
+
+BFF 函数定义的路由会在 `app.ts` 文件定义的路由之后注册，所以在这里你也可以拦截 BFF 函数定义的路由，进行预处理或是提前响应。
+
+:::note
+此时，如果应用中同时存在 `api/_app.ts`，则定义的中间件会被放在 `api/app.ts` 导出实例的最后执行。多数情况下，中间件就能覆盖大多数 BFF 函数的定制需求。只有当应用的服务端逻辑比较复杂时，才需要自定义 Server 实例。
+:::
+
+:::caution
+当应用中没有 `api/app.ts` 的时候，Modern.js 默认会添加 [koa-body](https://www.npmjs.com/package/koa-body)。当应用中存在 `api/app.ts` 时，如果开发者希望使用带有 Body 的 BFF 函数，需要**自行添加** koa-body。
```
Comment (yimingjfe) [inline packages/document/main-doc/docs/zh/guides/advanced-features/bff/extend-server.mdx:155]:
```suggestion
当应用中没有 `api/app.ts` 的时候，Modern.js 默认会添加 [koa-body](https://www.npmjs.com/package/koa-body)。当应用中存在 `api/app.ts` 时，如果开发者希望使用带有 Body 的 BFF 函数，需要**自行添加** `koa-body`。
```

File: packages/document/main-doc/docs/zh/guides/advanced-features/bff/function.mdx (hunk 1 above)
Comment (zllkjc) [reply packages/document/main-doc/docs/zh/guides/advanced-features/bff/function.mdx:126]: OK, i will remove it

File: packages/document/main-doc/docs/zh/guides/advanced-features/bff/function.mdx (hunk 2 above)
Comment (zllkjc) [reply packages/document/main-doc/docs/zh/guides/advanced-features/bff/function.mdx:3]: Get，我改成 api/lambda

**Review by yimingjfe — APPROVED** [review]

**Pull request merged**